schema.json

```diff
@@ -43,131 +43,142 @@
+                "minProperties": { "$ref": "#/definitions/positiveIntegerDefault0" },
+                "required": { "$ref": "#/definitions/stringArray" },
+                "additionalProperties": {
+                    "anyOf": [
+                        { "type": "boolean" },
+                        { "$ref": "#" }
+                    ],
+                    "default": {}
+                },
+                "definitions": {
+                    "type": "object",
+                    "additionalProperties": { "$ref": "#" },
+                    "default": {}
+                },
+                "properties": {
+                    "type": "object",
+                    "additionalProperties": { "$ref": "#" },
+                    "default": {}
+                },
+                "patternProperties": {
+                    "type": "object",
+                    "additionalProperties": { "$ref": "#" },
+                    "default": {}
+                },
+                "dependencies": {
+                    "type": "object",
+                    "additionalProperties": {
+                        "anyOf": [
+                            { "$ref": "#" },
+                            { "$ref": "#/definitions/stringArray" }
+                        ]
+                    }
+                },
+                "const": {},
+                "enum": {
                     "type": "array",
-                    "items": { "$ref": "#/definitions/simpleTypes" },
                     "minItems": 1,
                     "uniqueItems": true
+                },
+                "type": {
+                    "anyOf": [
+                        { "$ref": "#/definitions/simpleTypes" },
+                        {
+                            "type": "array",
+                            "items": { "$ref": "#/definitions/simpleTypes" },
+                            "minItems": 1,
+                            "uniqueItems": true
+                        }
+                    ]
+                },
+                "format": { "type": "string" },
+                "allOf": { "$ref": "#/definitions/schemaArray" },
+                "anyOf": { "$ref": "#/definitions/schemaArray" },
+                "oneOf": { "$ref": "#/definitions/schemaArray" },
+                "not": { "$ref": "#" }
+            },
+            "dependencies": {
+                "exclusiveMaximum": [ "maximum" ],
+                "exclusiveMinimum": [ "minimum" ]
+            },
+            "not": {
+                "properties": {
+                    "$ref": {}
                 }
-            ]
-        },
-        "format": { "type": "string" },
-        "allOf": { "$ref": "#/definitions/schemaArray" },
-        "anyOf": { "$ref": "#/definitions/schemaArray" },
-        "oneOf": { "$ref": "#/definitions/schemaArray" },
-        "not": { "$ref": "#" }
-    },
-    "dependencies": {
-        "exclusiveMaximum": [ "maximum" ],
-        "exclusiveMinimum": [ "minimum" ]
+            }
```

Comment:
> So that negated schema successfully validates any object that has a "$ref" property. 

Yes, that is correct. It however also successfully validates any object without "$ref" property.